projects/plugins/social/src/js/components/admin-page/index.jsx

```diff
@@ -63,13 +66,49 @@ const ConnectionItems = () => {
 		);
 	} );
 };
+
+const ModuleToggle = () => {
+	const updateOptions = useDispatch( STORE_ID ).updateJetpackSettings;
+	const isModuleEnabled = useSelect( select => select( STORE_ID ).isModuleEnabled() );
+	const isUpdating = useSelect( select => select( STORE_ID ).isUpdatingJetpackSettings() );
+
+	const toggleModule = useCallback( () => {
+		const newOption = {
+			publicize_active: ! isModuleEnabled,
+		};
+		updateOptions( newOption );
+	}, [ isModuleEnabled, updateOptions ] );
+
+	const label = isModuleEnabled
+		? __( 'Jetpack Social is active', 'jetpack-social' )
+		: __( 'Jetpack Social is inactive', 'jetpack-social' );
+
+	return (
+		<ToggleControl
+			label={ __( 'Activate Jetpack Social', 'jetpack-social' ) }
+			help={ isUpdating ? __( 'Updating…', 'jetpack-social' ) : label }
+			disabled={ isUpdating }
+			checked={ isModuleEnabled }
+			onChange={ toggleModule }
```

Comment:
In this case, it's because `onChange` is the callback [provided by the component](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/toggle-control). It renders an HTML checkbox, so I'm not sure if there would be an actual change in behavior when using `onClick` instead. Perhaps something to do with handling keyboard events etc.

In general, `onChange` fires when the value of an input field changes whereas `onClick` fires anytime the element gets clicked, regardless of a change in value.

Reply:
Ah, that makes sense!